packages/auth/graphql/resolvers/_mutations/clearSessions.js

```diff
@@ -0,0 +1,41 @@
+const t = require('../../../lib/t')
+const Roles = require('../../../lib/Roles')
+const { DestroySessionError } = require('../../../lib/errors')
+const ensureSignedIn = require('../../../lib/ensureSignedIn')
+const { clearAllUserSessions, destroySession } = require('../../../lib/Sessions')
+const userAccessRoles = ['admin', 'supporter']
+
+module.exports = async (_, args, { pgdb, user: me, req }) => {
+  ensureSignedIn(req)
+
+  const {
+    userId: foreignUserId
+  } = args
+
+  const user = foreignUserId
+    ? (await pgdb.public.users.findOne({ id: foreignUserId }))
+    : me
+
+  try {
+    let isSessionsCleared = false
+    if (me.id === user.id) {
+      // current user targeted, so we
+      // destroy the current session safely via express
+      if (await destroySession(req)) isSessionsCleared = true
```

Comment:
wouldn't set `isSessionsCleared` here yet. Is only truly done when all are cleared